migrations/2022-06-21-123144_language-tags/up.sql

```diff
@@ -12,9 +12,8 @@ create table local_user_language (
 
 alter table local_user rename column lang to interface_language;
 
-alter table post add column language_id integer references language not null default 0;
-
 insert into language(id, code, name) values (0, 'und', 'Undetermined');
+alter table post add column language_id integer references language not null default 0;
```

Comment:
There is something wrong with this migration, test servers are broken (check enterprise.lemmy.ml). I tried to fix it with this change (deployed on ds9), but it didnt help.

```
postgres_1  | 2022-08-24 11:42:07.218 UTC [47] ERROR:  insert or update on table "post" violates foreign key constraint "post_language_id_fkey"
postgres_1  | 2022-08-24 11:42:07.218 UTC [47] DETAIL:  Key (language_id)=(0) is not present in table "language".
postgres_1  | 2022-08-24 11:42:07.218 UTC [47] STATEMENT:  create table language (
postgres_1  |     id serial primary key,
postgres_1  |     code varchar(3),
postgres_1  |     name text
postgres_1  |   );
postgres_1  |   create table local_user_language (
postgres_1  |     id serial primary key,
postgres_1  |     local_user_id int references local_user on update cascade on delete cascade not null,
postgres_1  |     language_id int references language on update cascade on delete cascade not null,
postgres_1  |     unique (local_user_id, language_id)
postgres_1  |   );
postgres_1  | 
postgres_1  |   alter table local_user rename column lang to interface_language;
postgres_1  | 
postgres_1  |   alter table post add column language_id integer references language not null default 0;
postgres_1  | 
postgres_1  |   insert into language(id, code, name) values (0, 'und', 'Undetermined');
postgres_1  |   insert into language(code, name) values ('aa', 'Afaraf');
...
postgres_1  |   insert into language(code, name) values ('io', 'Ido');
lemmy_1     | Error: LemmyError { message: None, inner: Failed with: insert or update on table "post" violates foreign key constraint "post_language_id_fkey", context: "SpanTrace" }
postgres_1  |   insert into language(code, name) values ('is', 'Íslenska');
...
```

Reply:
It says it ran the two migrations, but the language table isn't there. Not sure how that happened, but you might have to either copy-paste those migration lines manually, or delete those two rows and try again.

```
lemmy=# select * from __diesel_schema_migrations ;

...
20220612012121 | 2022-08-17 01:00:28.9136                                                                                20220804214722 | 2022-08-18 01:00:29.931373                                                                             (118 rows)
                                                                                                                         
```

Reply:
Are you checking on the server? I guess its because the migration was reverted after it threw an error. But based on the logs, this seems to be the root cause: `Key (language_id)=(0) is not present in table "language".`

Reply:
Yep that was from the server. The language table is missing, yet postgres says it ran that migration that includes the language table, which is why its giving that error. The only fix is to either delete those rows from that migrations table, so it'll run the migration again, or run the code in the migrations manually. 

No idea how it ended up in that weird state.

Reply:
I now manually deleted the rows from __diesel_schema_migrations and reverted migrations. Now it seems like the migration ran without problems, but the site is still down. Lemmy-ui logs "request to http://0.0.0.0:8536/api/v3/site? failed, reason: connect ECONNREFUSED 0.0.0.0:8536", even after multiple restarts.